From 550e26297f7277689cb9ee37c983b2692ff981e9 Mon Sep 17 00:00:00 2001
From: Li Guohui <liguohui2061@phytium.com.cn>
Date: Thu, 16 May 2024 16:33:54 +0800
Subject: [PATCH 053/360] gpio: phytium: Add old gpio-ports index description
 to adapt old firmware

The recently patch for GPIO changed the description of gpio-ports index
in gpio-plat drivers, which may cause the driver cannot property read
old description in the old ACPI table.

Signed-off-by: Li Guohui <liguohui2061@phytium.com.cn>
Signed-off-by: Lan Hengyu <lanhengyu1395@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
---
 drivers/gpio/gpio-phytium-platform.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpio/gpio-phytium-platform.c b/drivers/gpio/gpio-phytium-platform.c
index 40a36668eba7..6c5d3b0a4bbb 100644
--- a/drivers/gpio/gpio-phytium-platform.c
+++ b/drivers/gpio/gpio-phytium-platform.c
@@ -77,8 +77,10 @@ static int phytium_gpio_probe(struct platform_device *pdev)
 			return -EINVAL;
 		}
 
-		if (fwnode_property_read_u32(fwnode, "nr-gpios",
-					     &gpio->ngpio[idx])) {
+		if (fwnode_property_read_u32(fwnode, "ngpios",
+					&gpio->ngpio[idx])
+				&& fwnode_property_read_u32(fwnode, "nr-gpios",
+					&gpio->ngpio[idx])) {
 			dev_info(dev,
 				 "failed to get number of gpios for Port%c\n",
 				 idx ? 'B' : 'A');
-- 
2.20.1

